### PR TITLE
Write a document for custom formatter of migrate

### PR DESCRIPTION
**Pull request by iwata**
SSIA
Related to this issue #2408.

### CODE REVIEW

File: doc/md/versioned-migrations.md [hunk 1]
```diff
@@ -174,3 +174,40 @@ versioned migration, you need to take some extra steps.
    In case of `golang-migrate` this can be done by forcing your database version as
    described [here](https://github.com/golang-migrate/migrate/blob/master/GETTING_STARTED.md#forcing-your-database-version).
 
+## Use a custom formatter
```
Comment (masseelch) [inline doc/md/versioned-migrations.md:177]:
```suggestion
## Use a Custom Formatter
```

File: doc/md/versioned-migrations.md [hunk 2]
```diff
@@ -174,3 +174,40 @@ versioned migration, you need to take some extra steps.
...
+
+It is possible to use another tool for migration management instead of `golang-migrate/migrate`. For example, you can define and use the following formatter to use [pressly/goose](https://github.com/pressly/goose).
```
Comment (masseelch) [inline doc/md/versioned-migrations.md:179]:
```suggestion
Atlas' migration engine comes with great customizability. By the use of a custom `Formatter` you can generate the migration files in a format compatible with another tool for migration management: [pressly/goose](https://github.com/pressly/goose).
```

File: doc/md/versioned-migrations.md [hunk 3]
```diff
@@ -174,3 +174,40 @@ versioned migration, you need to take some extra steps.
...
+
+```go
+var (
+	templateFuncs = template.FuncMap{
+		"now": time.Now,
+		"sem": ensureSemicolonSuffix,
+		"rev": reverse,
+	}
+	// gooseFormatter is an implementation for compatible formatter with goose.
+	// https://github.com/pressly/goose/blob/205faf5cd314dcc76f2687bac7cfe0534a4bd5a9/create.go#L90-L99
```
Comment (masseelch) [inline doc/md/versioned-migrations.md:189]: I am not sure, why you link that part of the repo. Do you think it is needed?

**Review by masseelch — CHANGES_REQUESTED** [review]
Hey @iwata, great start, thank you! 

I added some minor comments. Additionally, I think we should try to give the users more ready-to-copy code to reduce friction when using a custom formatter.

I think a good idea would be to copy [this](http://localhost:3000/docs/versioned-migrations#from-graph) code snippet and add the changes needed for the `gooseFormatter` and [highlight them](https://docusaurus.io/docs/markdown-features/code-blocks#line-highlighting), WDYT? .

File: doc/md/versioned-migrations.md (hunk 3 above)
Comment (iwata) [reply doc/md/versioned-migrations.md:189]: I wrote this just in case the format definition for original is in the link. Perhaps it is not necessary.

**Comment by iwata** [comment]
@masseelch

> I added some minor comments. Additionally, I think we should try to give the users more ready-to-copy code to reduce friction when using a custom formatted.

Thanks, and committed to your suggestions.


> I think a good idea would be to copy [this](http://localhost:3000/docs/versioned-migrations#from-graph) code snippet and add the changes needed for the gooseFormatter and [highlight them](https://docusaurus.io/docs/markdown-features/code-blocks#line-highlighting), WDYT? .

GOTCHA 👍 
I've updated the code sample, so PTAL.

**Comment by iwata** [comment]
@masseelch Can you check this?

**Comment by masseelch** [comment]
> @masseelch Can you check this?

Yes, of course. Somehow this slipped my attention, sorry!

File: doc/md/versioned-migrations.md [hunk 4]
```diff
@@ -174,3 +174,111 @@ versioned migration, you need to take some extra steps.
...
+    if err != nil {
+        log.Fatalln(err)
+    }
+    if err := m.Diff(context.Background(), tbls...); err != nil {
+        log.Fatalln(err)
+    }
+    // You can use the following method to give the migration files a name.
+    // if err := m.NamedDiff(context.Background(), "migration_name", tbls...); err != nil {
+    //     log.Fatalln(err)
+    // }
```
Comment (masseelch) [inline doc/md/versioned-migrations.md:282]:
```suggestion
```

**Review by masseelch — APPROVED** [review]
Will merge after those suggestions :-) Thanks!

File: doc/md/versioned-migrations.md [hunk 5]
```diff
@@ -174,3 +174,111 @@ versioned migration, you need to take some extra steps.
...
+	}
+	return s
+}
+
+func init() {
+    // highlight-start
+    // Replace the formatter to output migration files compatible with `goose`.
+    migrate.DefaultFormatter = gooseFormatter
+    // highlight-end
+}
```
[the remark this answers is not in the document]
Comment (iwata) [reply doc/md/versioned-migrations.md:249]: Why? I think this is required to customize the formatter.

File: doc/md/versioned-migrations.md [hunk 6]
```diff
@@ -174,3 +174,107 @@ versioned migration, you need to take some extra steps.
...
+    if err != nil {
+        log.Fatalln(err)
+    }
+    // Open connection to the database.
+    dlct, err := sql.Open("mysql", "root:pass@tcp(localhost:3306)/test")
+    if err != nil {
+        log.Fatalln(err)
+    }
+    // Inspect it and compare it with the graph.
+    m, err := schema.NewMigrate(dlct, schema.WithDir(d))
```
Comment (masseelch) [inline doc/md/versioned-migrations.md:272]:
```suggestion
    // highlight-start
    m, err := schema.NewMigrate(dlct, schema.WithDir(d), schema.WithFormatter(goose formatter))
    // highlight-end
```

**Review by masseelch — COMMENTED** [review]
Forgot one part. That is why I removed the first one

**Comment by iwata** [comment]
> Forgot one part. That is why I removed the first one

Ah hah, I finally understand 😄

**Review by masseelch — APPROVED** [review]

**Pull request merged**